pages/api/balance/[address].ts

```diff
@@ -0,0 +1,49 @@
+import { NextApiResponse, NextApiRequest } from 'next'
+import { getBCHBalance } from 'services/grpcService'
+import { RESPONSE_MESSAGES } from 'constants/index'
+import { parseAddress } from 'utils/validators'
+import Cors from 'cors'
+import { satoshisToUnit } from 'utils'
+import xecaddr from 'xecaddrjs'
+import { Decimal } from '@prisma/client/runtime'
+
+const { ADDRESS_NOT_PROVIDED_400 } = RESPONSE_MESSAGES
+const cors = Cors({
+  methods: ['GET', 'HEAD']
+})
+
+async function runMiddleware (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  fn: Function
+): Promise<void> {
+  return await new Promise((resolve, reject) => {
+    fn(req, res, (result: any) => {
+      if (result instanceof Error) {
+        return reject(result)
+      }
+
+      return resolve(result)
+    })
+  })
+}
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  await runMiddleware(req, res, cors)
+  if (req.method === 'GET') {
+    try {
+      const address = parseAddress(req.query.address as string)
+      const response = await getBCHBalance(address)
+      const balance = satoshisToUnit(new Decimal(response), xecaddr.Format.Cashaddr)
```

Comment:
`await` statement missing here, making the response return an empty JSON instead of the balance